### DISCUSSION TITLE
AR rendering quality difference Android vs Apple

### DISCUSSION THREAD

**Discussion by ABakker30** [google/model-viewer#3642] [Q&A]
User feedback indicates better AR Apple rendering results than Android for the same USDZ/GLTF file.
Several recorded Apple AR rendering examples: https://bit.ly/37zp5Mc, https://bit.ly/3aulvF5 and https://bit.ly/3rdHEgT
Several recorded Android AR rendering examples: https://bit.ly/2LUbXsN , https://bit.ly/38h6N1u and https://bit.ly/34vsNEK

Unless I missed something, I'm not aware of any setting I could alter to improve the Android results.

If someone would like to do their own testing respond with a location pin and I'll add it to the GEO-AR locations on this site with a 500 meter radius: https://www.globalsculpturepark.com/

**Comment by elalish** [comment]
It looks like your comparing scene-viewer to quick-look, neither of which are part of this code base. It would be helpful if you could describe exactly what is better or worse; I'm guessing the main problems are that scene-viewer is less shiny and the reflections change suddenly. Have you tried WebXR mode (our default on Android)? This is the rendering we control here and I'd like to know how you think it compares to these other two.

**Comment by canberka** [comment]
this is issue we realised .. and it can be  big problem for customers. Is there any solution to fix it ? 
![WhatsApp Image 2021-03-03 at 16 28 38](https://user-images.githubusercontent.com/36857390/109813559-9421ed00-7c3e-11eb-8273-58f8d36049b6.jpeg)
![WhatsApp Image 2021-03-03 at 16 30 19](https://user-images.githubusercontent.com/36857390/109813565-95ebb080-7c3e-11eb-890b-ba8e3fb5e6de.jpeg)

**Comment by elalish** [comment]
Can you answer the questions I asked above? I have no idea which modes your images relate to.

**Comment by ABakker30** [comment]
Sorry for the late follow-up....

Based on your suggestion I did try webxr without significant improvement. A recent White House / Washington Post article drove 30,000 views during the last week. It is hard to describe the clear quality differences. It may be better to give you two Google photo album links one with Apple AR images and one with Android images for your own comparison. The images are based on the same models/textures maps. 

Apple images link: https://photos.app.goo.gl/YeugxghVDNKKKao66
Android images link: https://photos.app.goo.gl/3cLZLdUryWshBVDV8

I would focus on comparing the Metal/Steel looking sculptures. Apple will look significantly better than Android

Other comments we received are related to clipping planes with Android. No such issues have been reported with Apple.

Also we have noted that Android phones loose tracking of the object when the camera is pointed upwards to look at the top of sculptures. This is not the case for Apple. Maybe this is related to a gimbal lock type issue. 

Overall the technology works fabulously well! We received great feedback from users.

There appears to be small change that we may be presenting this at the South Lawn of the White House thanks to the article in the post: https://www.washingtonpost.com/goingoutguide/museums/anton-bakker-art-review/2021/02/23/e66a6a00-71fe-11eb-93be-c10813e358a2_story.html

Let me know if I can be of any further help,

**Comment by elalish** [comment]
Thank you! I love your generative sculptures; I've done some mathematical sculpture [myself](https://www.thingiverse.com/emmett/designs), so it's right up my alley. 

So, you say WebXR didn't make much difference, but can you share some similar screenshots? I say this because I'm pretty certain I know exactly the sceneViewer (really Filament) bug that is causes these shiny sculptures to look more matte. It's an internal clamp on the minimum roughness. However, there's also differences in lighting; ARCore and ARKit both estimate the lighting environment you're in, but do so differently, which makes large changes in rendering.

Finally, the big value add of glTF is in standardization of rendering, which we test [here](https://modelviewer.dev/fidelity/) for instance. There's nothing of this kind for conversion of glTF to USDZ or for Apple's renderers, so the differences are likely to be larger. Out of curiosity, did you create glTFs and convert them to USDZ or vice-versa?

As for clipping and losing track, those sound more like ARCore issues. If you can be more specific and give screenshots/recordings, I'll be happy to pass them along to that team.

**Comment by elalish** [comment]
One other thing; in WebXR we don't estimate lighting yet, so we'll use whatever you apply even in AR. You may want to test with an outdoor, sunny HDR environment to see what difference that lighting makes.

**Comment by ABakker30** [comment]
Thanks for liking the work :) I do make them as physical works as well. This is a good one for Google: https://ar-view.com/viewer/artwork/296e6676-5896-4db4-90e4-11c9badc9eae  The project is a year long virtual project with a monthly new collection. The subjects are math related and educational in nature. Do you have anyone within Google that would be interested in discussing the AR/VR project?  

Regarding your questions:

Video recording of the clipping issue: https://photos.app.goo.gl/mMLwt3H1mbW9MtQS7

I do not have an example of the loosing track when pointing the camera to the sky, easy to simulate with any of the sculptures here: www.globalsculpturepark.com Enlarge the sculptures to 30 or so feet tall, walk underneath the sculptures and look to the top. It is likely to loose tracking every time. Apple has never lost tracking.

We generate GLTF and USDZ separately using Substance ​Painter.

Let me know if can can provide any additional info, I love this technology.

**Comment by elalish** [comment]
Thanks for the details on the clipping issue; @tpsiaki to take a look. 

What do you want to talk to Google about? I'm certainly interested, and I also have some contacts at Google Arts & Culture that might be more what you're looking for?

**Comment by ABakker30** [comment]
I would like to promote the technology and its capabilities including educational, cultural, etc. value. Would love to add a powered by Google... on the page. How could we raise the interest for VR? For virtual museums like these: https://www.antonbakker.com/momath
Maybe take this offline
